apps/backend/src/auth/auth.service.ts

```diff
@@ -14,10 +15,47 @@ export default new Elysia()
     ConflictError,
     UnauthorizedError,
   })
-  .derive(async ({ jwt }) => ({
+  .derive(async ({ accessJwt, refreshJwt }) => ({
     authService: {
-      signToken: async (user: { userId: string; username: string }) => {
-        return jwt.sign({ id: user.userId, sub: user.username });
+      signTokenPair: async (user: { userId: string; username: string }) => {
+        // TODO: Access token should live much shorter than refresh token
+        const accessToken = await accessJwt.sign({
+          id: user.userId,
+          sub: user.username,
+          typ: TokenType.Access,
+        });
+        const refreshToken = await refreshJwt.sign({
+          id: user.userId,
+          sub: user.username,
+          typ: TokenType.Refresh,
+        });
+        // Invalidate all valid refresh tokens for this user
+        await prisma.refreshToken.updateMany({
```

Comment:
Do we need to actually invalidate all the tokens? Also, this is some weird duplication of logic [2]. This could be extracted to a function so we don't do this twice.

Reply:
I extracted the token invalidation logic to a utils module. We're still doing 2 selects, 1 update and 1 insert per token refresh tho. One select could probably be removed, but we're storing usernames in tokens, so we have to query the database for it.